kubernetes-model-generator/kubernetes-model-core/src/main/java/io/fabric8/kubernetes/api/model/GenericKubernetesResource.java

```diff
@@ -69,4 +77,54 @@ public void setAdditionalProperty(String name, Object value) {
     this.additionalProperties.put(name, value);
   }
 
+  /**
+   * Allows the retrieval of field values from this Resource for the provided path segments.
+   *
+   * <p> The following notation is supported
+   * <ul>
+   *   <li>
+   *     Dot Notation
+   *     <pre>{@code resource.get("path.of.the.nested.prop");}</pre>
+   *   </li>
+   *   <li>
+   *     Varargs Notation
+   *     <pre>{@code resource.get("path", "of", "the", "nested", "prop");}</pre>
+   *   </li>
+   *   <li>
+   *     Mixed Notation
+   *     <pre>{@code resource.get("path.of", "the.nested", "prop");}</pre>
+   *   </li>
+   * </ul>
+   *
+   * <p> In addition, fields that contain collections/arrays can also be traversed
+   * <pre>{@code resource.get("path.collectionField[1].nested.array[0].field");}</pre>
+   *
+   * @param path of the field to retrieve.
+   * @param <T> type of the returned object.
+   * @return the value of the traversed path or null if the field does not exist.
+   */
+  @SuppressWarnings("unchecked")
+  public <T> T get(String... path) {
+    Object current = null;
+    int it = 0;
```

Comment:
Maybe could be changed to `boolean isFirstTime;` I understand that an integer is easy to manage as you do a `++` in the conditional. Maybe leave it as integer but with meaningful name?

Reply:
No longer applicable after considering https://github.com/fabric8io/kubernetes-client/pull/3619#pullrequestreview-816316524